pkg/clusteroperator/clusteroperators.go

```diff
@@ -0,0 +1,179 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"fmt"
+)
+
+// CoListBuilder provides struct for ClusterOperator List.
+type CoListBuilder struct {
+	// ClusterOperatorList definition. Used to create a clusteroperator object list
+	Definition []*ClusterOperatorBuilder
+	// Created clusteroperator object list
+	ObjectList []*ClusterOperatorBuilder
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Discover method gets the clusterOperators list in cluster and stores them in the builder struct.
+func (builder *CoListBuilder) Discover() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Getting the list of clusterOperator objects on this cluster")
+
+	builder.ObjectList = nil
+
+	coList, err := builder.apiClient.ClusterOperators().List(
+		context.TODO(), metaV1.ListOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Error to list clusterOperators")
+
+		return err
+	}
+
+	if len(coList.Items) < 1 {
+		glog.V(100).Infof("Cluster doesn't have clusterOperator deployed, bad cluster state ...")
+
+		return fmt.Errorf("clusterOperator list is empty")
+	}
+
+	for _, clusterOperator := range coList.Items {
+		copiedCo := clusterOperator
+		coBuilder := &ClusterOperatorBuilder{
+			apiClient:  builder.apiClient,
+			Object:     &copiedCo,
+			Definition: &copiedCo,
+		}
+
+		builder.ObjectList = append(builder.ObjectList, coBuilder)
+	}
+
+	return err
+}
+
+// WaitToBeAllAvailableFor waits for all clusterConfigs to be available for a time duration up to the timeout.
+func (builder *CoListBuilder) WaitToBeAllAvailableFor(timeout time.Duration) (bool, error) {
+	glog.V(100).Infof("Wait for each clusterOperator from the clusterOperatorsList to reach "+
+		"Available state up to duration of %v", timeout)
+
+	isAllCoAvailable := true
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out
+	// after timeout
+	err := wait.PollImmediate(fiveScds, timeout, func() (bool, error) {
+		err := builder.Discover()
+
+		if err != nil {
+			return false, err
+		}
+
+		// iterate through the ClusterOperators in the list.
+		for _, clusteroperator := range builder.ObjectList {
```

Comment:
Sorry if I accidentally caused some confusion, I think you should actually run the wait for each clusteroperator similar to this:
```
for _, clusteroperator := range builder.ObjectList {
  err := wait.PollImmediate(fiveScds, timeout, func() (bool, error) {
    return clusteroperator.IsAvailable(), nil
  }

  if err != nil {
    glog.V(100).Infof("Not all clusterOperators were found available during availableDuration: %v",
		timeout)

    return false, err
  }

}

glog.V(100).Infof("All clusterOperators were found available during availableDuration: %v",
			timeout)

return true, nil
```

This mimics the same behavior as the `oc wait` command